signac_dashboard/modules/navigator.py

```diff
@@ -0,0 +1,99 @@
+from flask import render_template, url_for
+
+from signac_dashboard.module import Module
+from signac_dashboard.util import escape_truncated_values
+
+
+class Navigator(Module):
+    """Displays links to jobs differing in one state point parameter.
+
+    This module may not update if the signac project changes because the module
+    caches the project schema when the signac-dashboard launches.
+
+    :param context: Supports :code:`'JobContext`
+    :type context: str
+    """
+
+    _supported_contexts = {"JobContext"}
+
+    def __init__(
+        self,
+        name="Navigator",
+        context="JobContext",
+        template="cards/navigator.html",
+        **kwargs,
+    ):
+        super().__init__(name=name, context=context, template=template, **kwargs)
+
+    def get_cards(self, job):
+        project = self._dashboard.project
+
+        def link_label(key, other_val):
+            """Returns the url and label for the job with job.sp[key] == other_val."""
+
+            similar_sp = job.sp()  # modifiable
+            similar_sp.update({key: other_val})
+
+            # Look only for exact matches of only changing one parameter
+            # in case of heterogeneous schema
+            other_job = project.open_job(similar_sp)
+            if other_job in project:
+                link = url_for("show_job", jobid=other_job.id)
+                label = other_job.sp[key]
+            else:
+                link = None
+                label = "no match"
+            return link, label
+
+        nearby_jobs = dict()
+        # for each parameter in the schema, find the next and previous job and get links to them
+        for key, values in self._sorted_schema.items():
+            my_val = job.sp.get(key, None)
+            if my_val is None:
+                # Possible if schema is heterogeneous
+                continue
+
+            my_index = values.index(my_val)
```

Comment:
We tend to avoid names like `my_index` in favor of something descriptive -- or just a simpler name. The `my_` prefix doesn't add meaning and doesn't really help with disambiguation. `val` for value and `idx` for index are fine.